docs/3.0rc/develop/results.mdx

```diff
@@ -1,546 +1,171 @@
-
-```python
-from prefect import flow, task
-
-@task
-def my_task():
-    raise ValueError()
-
-@flow
-def my_flow():
-    try:
-        my_task()
-    except ValueError:
-        print("Oh no! The task failed.")
-
-    return True
-
-my_flow()
-```
-
-If you prefer to check for a failed task without using `try/except`, you may ask Prefect to return the state:
-
-```python
-from prefect import flow, task
-
-@task
-def my_task():
-    raise ValueError()
-
-@flow
-def my_flow():
-    state = my_task(return_state=True)
-
-    if state.is_failed():
-        print("Oh no! The task failed. Falling back to '1'.")
-        result = 1
-    else:
-        result = state.result()
-
-    return result + 1
-
-result = my_flow()
-assert result == 2
-```
-
-If you retrieve the result from a failed state, the exception is raised. 
-For this reason, it's often best to check if the state is failed first.
-
-```python
-from prefect import flow, task
-
-@task
-def my_task():
-    raise ValueError()
-
-@flow
-def my_flow():
-    state = my_task(return_state=True)
-
-    try:
-        result = state.result()
-    except ValueError:
-        print("Oh no! The state raised the error!")
-
-    return True
-
-my_flow()
-```
-
-When retrieving the result from a state, you can ask Prefect not to raise exceptions:
-
-```python
-from prefect import flow, task
-
-@task
-def my_task():
-    raise ValueError()
-
-@flow
-def my_flow():
-    state = my_task(return_state=True)
-
-    maybe_result = state.result(raise_on_failure=False)
-    if isinstance(maybe_result, ValueError):
-        print("Oh no! The task failed. Falling back to '1'.")
-        result = 1
-    else:
-        result = maybe_result
-
-    return result + 1
-
-result = my_flow()
-assert result == 2
-```
-
-When submitting tasks to a runner, `Future.result()` works the same as `State.result()`:
-
-```python
-from prefect import flow, task
-
-@task
-def my_task():
-    raise ValueError()
-
-@flow
-def my_flow():
-    future = my_task.submit()
-
-    try:
-        future.result()
-    except ValueError:
-        print("Ah! Futures will raise the failure as well.")
-
-    # You can ask it not to raise the exception too
-    maybe_result = future.result(raise_on_failure=False)
-    print(f"Got {type(maybe_result)}")
-
-    return True
-
-my_flow()
-```
-
-### Working with async results
-
-When **calling** flows or tasks, the result is returned directly:
+<Tip>
+**Turn on persistence globally by default**
 
-```python
-import asyncio
-from prefect import flow, task
+The simplest way to turn on result persistence globally is through the `PREFECT_RESULTS_PERSIST_BY_DEFAULT` setting:
 
-@task
-async def my_task():
-    return 1
-
-@flow
-async def my_flow():
-    task_result = await my_task()
-    return task_result + 1
-
-result = asyncio.run(my_flow())
-assert result == 2
+```bash
+prefect config set PREFECT_RESULTS_PERSIST_BY_DEFAULT=true
```

Comment:
`PREFECT_DEFAULT_RESULT_STORAGE_BLOCK` This is still not mentioned -- can we either add it or add it to the page we link to (/3.0rc/manage/settings-and-profiles)

It gets asked for a lot so I would want this to be more findable in the docs

Reply:
Yup absolutely - let me merge these updates as-is just to get the foundation in, and I promise to revisit this week and add in more on that setting 👍